src/components/posts/AddPostForm.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react'
+import Textarea from '../textarea/Textarea'
+import { useSetRecoilState } from 'recoil'
+import postAtom from './State'
+
+const AddPostForm = () => {
+  const [textContent, setTextContent] = useState<string>('')
+  const setContentPost = useSetRecoilState(postAtom)
+
+  useEffect(() => {
+    textContent &&
+      setContentPost(previous => [
```

Comment:
Improvement suggestion. You're now using a local state of `textContent` specifically to add a new post when it is populated via this effect. Why not use a callback `addPost` that does this instead of the effect and pass that to the `Textarea` component?